clang/include/clang/Frontend/FrontendAction.h

```diff
@@ -276,12 +276,18 @@ class PluginASTAction : public ASTFrontendAction {
                              ///< the command line
     ReplaceAction,           ///< Replace the main action
     AddBeforeMainAction,     ///< Execute the action before the main action
-    AddAfterMainAction       ///< Execute the action after the main action
+    AddAfterMainAction,      ///< Execute the action after the main action
+    ReplaceAndReuseAction,   ///< Replace the main action by an action that reuses
+                             ///< the original main action
   };
   /// Get the action type for this plugin
   ///
   /// \return The action type. By default we use CmdlineAfterMainAction.
   virtual ActionType getActionType() { return CmdlineAfterMainAction; }
+
+  /// Iff the Action is of ReplaceAndReuseAction the main action to replace will
```

Comment:
```suggestion
  /// If the Action is ReplaceAndReuseAction, the main action to replace will
```
I guess you did not mean the mathematical `iff`, or did you?

Reply:
This is iff in the mathematical sense. `takeMainActionToReplace` is not overloaded if there Action type is not `ReplaceAndReuseAction`